tests/test_query.js

```diff
@@ -92,6 +105,35 @@ test('query() matches a query string using regexp', t => {
   })
 })
 
+test('query() accepts URLSearchParams as input', async t => {
+  const params = new url.URLSearchParams({
+    foo: 'bar',
+  })
+
+  const scope = nock('http://example.test')
+    .get('/')
+    .query(params)
+    .reply()
+
+  const { statusCode } = await got('http://example.test?foo=bar')
+
+  t.is(statusCode, 200)
+  scope.done()
+})
+
+test('multiple set query keys use the first occurrence', async t => {
```

Comment:
Seems like using the last occurrence might make more sense… or perhaps it would be better to throw an error?

I've noticed that when there are multiple calls to `reply()`, one of them takes preference… and come to think of it, I forget which. Throwing an error seems nicer to me.

Reply:
I was thinking the same thing. Since any change would be a breaking change, I thought it best to just add coverage of current expected functionality. I'll create an issue to add a breaking change and throw instead.

Reply:
#1623

Reply:
For what it's worth, I'm not sure if people are ever deliberately using the library that way. It's not a documented behavior. So I'm not sure if it's a breaking change or a feature.

If we can ship it in 11.0, so much the better!